clients/src/main/java/org/apache/kafka/common/security/scram/ScramSaslServer.java

```diff
@@ -90,6 +90,10 @@ public ScramSaslServer(ScramMechanism mechanism, Map<String, ?> props, CallbackH
                         this.scramCredential = credentialCallback.scramCredential();
                         if (scramCredential == null)
                             throw new SaslException("Authentication failed: Invalid user credentials");
+                        String authorizationIDFromClient = clientFirstMessage.authorizationId();
```

Comment:
A couple of questions:
1. Do we need to store `serverNonce` as a field?
2. Should we clear `clientFirstMessage` and `serverFirstMessage` after they're used?

Reply:
Made `serverNonce` a local variable, reset `scramCredentials`, `clientFirstMessage` and `serverFirstMessage` after use.